logstash-core/lib/logstash/output_delegator.rb

```diff
@@ -38,9 +89,16 @@ def register
     @workers.each {|w| w.register}
   end
 
-  def multi_receive(events)
+  def threadsafe_multi_receive(events)
+    @events_received.increment(events.length)
+
+    @threadsafe_worker.multi_receive(events)
+  end
+
+  def worker_multi_receive(events)
     @events_received.increment(events.length)
 
+    @logger.debug("worker queue pop")
```

Comment:
since this hasn't been removed yet, just adding a comment to ensure it wasn't forgotten :)